abci/types/types.proto

```diff
@@ -81,8 +81,15 @@ message RequestBeginBlock {
   repeated Evidence byzantine_validators = 4 [(gogoproto.nullable)=false];
 }
 
+enum CheckTxType {
+  Unchecked = 0;
+  Checked = 1;
+}
+
 message RequestCheckTx {
   bytes tx = 1;
+  CheckTxType type = 2;
+  bytes additional_data = 3;
```

Comment:
Is it possible for `additional_data` to introduce new security threats? If so, how would we go about mitigating them?